app/src/main/java/com/github/geohunt/app/ui/components/user/ProfileIcon.kt

```diff
@@ -1,33 +1,37 @@
 package com.github.geohunt.app.ui.components.user
 
+import androidx.compose.foundation.layout.aspectRatio
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.shape.CircleShape
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.unit.dp
 import coil.compose.AsyncImage
 import coil.request.ImageRequest
-import coil.size.Size
 import com.github.geohunt.app.model.database.api.User
+import com.github.geohunt.app.ui.rememberLazyRef
 
 /**
  * Creates a round profile icon of the given size.
  *
  * @param user The user for which we draw the profile picture.
- * @param size The size of the icon.
  */
 @Composable
-fun ProfileIcon(user: User, size: Size = Size.ORIGINAL) {
+fun ProfileIcon(user: User) {
+    val profilePicture = rememberLazyRef { user.profilePicture }
```

Comment:
Your code does not handle well null values. If user.profilePicture is null, your code will throw a NullPointerException. 

I would consider changing your code to look in the following way:

```
@Composable
fun ProfileIcon(user: User) {
    val profilePicture = rememberLazyRef { user.profilePicture }

    val imageModel = if (profilePicture.value != null) {
        ImageRequest.Builder(LocalContext.current)
            .data(profilePicture.value)
            .crossfade(true)
            .build()
    } else {
        // Handle the case where user or user.profilePicture is null
        // You can return a placeholder image, display a default icon, or show an error message, for example
        // For example simply return an empty composable
        Spacer(modifier = Modifier
            .aspectRatio(1f)
            .padding(8.dp)
            .clip(CircleShape)
            .background(Color.LightGray))
        return
    }

    AsyncImage(
        model = imageModel,
        contentDescription = "${user.name ?: "User"} profile picture",
        contentScale = ContentScale.Crop,
        modifier = Modifier
            .aspectRatio(1f)
            .padding(8.dp)
            .clip(CircleShape)
    )
}
```

This way you could handle the well cases with null values and not just crash the application with NullPointerException

Reply:
[Coil](https://coil-kt.github.io/coil/) (the library used for loading images) actually handles null value, and doesn't display anything in this case, so I don't understand your point with crashes and NPEs.